Library/Formula/slimerjs.rb

```diff
@@ -1,3 +1,23 @@
+class FirefoxDependency < Requirement
+  fatal true
+  cask "Firefox"
+  download "https://www.mozilla.org/en-US/firefox/new"
+
+  satisfy { firefoxappdira? || firefoxappdirb? }
+
+  def firefoxappdira?
+    File.exist?("~/Applications/Firefox.app")
+  end
```

Comment:
You have to expand `~` first:

```rb
irb(main):001:0> File.expand_path("~/Applications")
=> "/Users/jacknagel/Applications"
```

`File.exist?` on a symlink reports on the target of the symlink (i.e., it calls `stat` and not `lstat`).